### PR TITLE
Fix operand processing issues

### PR DESCRIPTION
**Pull request by riyafa**
Related issue: https://github.com/ballerina-platform/ballerina-lang/issues/28893

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java [hunk 1]
```diff
@@ -44,58 +44,61 @@
...
-            mainArgs = new Object[size];
             processOperands(option.getOperandArgs());
-            mainArgs[size - 2] = record;
-            mainArgs[size - 1] = true;
+            mainArgs.add(record);
+            mainArgs.add(true);
         } else {
-            mainArgs = new Object[operands.length * 2 + 1];
-            // Todo: create an empty option and call option.getOperandArgs()
             Module module = new Module("$anon", ".", "0.0.0");
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java:67]: We can make this dummy module a constant. IMO the correct way would be to pass the root module from codegen to CLispec and use that. ( We already done that in config passing)

File: bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java [hunk 2]
```diff
@@ -18,13 +18,193 @@
...
+        String val = "arg";
+        Object[] args = new CliSpec(null, operands, val).getMainArgs();
+        Assert.assertTrue(args[1] instanceof BString);
+        Assert.assertEquals(((BString) args[1]).getValue(), val);
+        Assert.assertTrue(args[3] instanceof BArray);
+        Assert.assertEquals(((BArray) args[3]).size(), 0);
+    }
+
+    @Test
+    public void testStringUnionNil() {
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java:199]: Do we have tests for rest params with value and without values?

File: bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java [hunk 3]
```diff
@@ -18,13 +18,193 @@
...
+import java.math.BigDecimal;
+
 /**
  * Tests the options feature in bal cli commands.
  */
 public class OperandTest {
     @Test
-    public void testOperand() {
+    public void testString() {
+        Operand[] operands = {new Operand(false, "opString", PredefinedTypes.TYPE_STRING)};
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java:40]: I think we can simply replace these with a data provider.

Comment (riyafa) [reply bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java:40]: The assertion part differs for each of the different types

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java (hunk 1 above)
Comment (riyafa) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java]: Fixed by passsing `null` for module as this is a dummy record

File: bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java (hunk 2 above)
Comment (riyafa) [reply bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java:199]: Added tests as jballerina-unit-tests

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java [hunk 4]
```diff
@@ -44,81 +43,92 @@
...
-            mainArgs[size - 2] = record;
-            mainArgs[size - 1] = true;
+            mainArgs.add(record);
+            mainArgs.add(true);
         } else {
-            mainArgs = new Object[operands.length * 2 + 1];
-            // Todo: create an empty option and call option.getOperandArgs()
-            Module module = new Module("$anon", ".", "0.0.0");
-            RecordType type = TypeCreator.createRecordType("dummy", module, 1, new HashMap<>(), null, true, 6);
+            RecordType type = TypeCreator.createRecordType("dummy", null, 1, new HashMap<>(), null, true, 6);
```
Comment (manuranga) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java:65]: `new HashMap<>()` shall we use Collections empty map?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliUtil.java [hunk 5]
```diff
@@ -119,4 +130,9 @@ private static DecimalValue getDecimalValue(String argument) {
                     StringUtils.fromString(String.format(INVALID_ARGUMENT_ERROR, argument, "decimal")));
         }
     }
+
+    static boolean isSupportedType(int tag) {
+        return tag == TypeTags.STRING_TAG || tag == TypeTags.INT_TAG || tag == TypeTags.FLOAT_TAG ||
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliUtil.java:135]: tag <= TypeTags.BOOLEAN_TAG

**Pull request closed without merging**